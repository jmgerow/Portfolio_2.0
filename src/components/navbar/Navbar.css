.navbar-fixed {
    z-index: 1;
}

