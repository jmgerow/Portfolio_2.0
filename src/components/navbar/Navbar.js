import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Navbar extends Component {
  constructor() {
    super();

    this.state = {
      title: "Justin Gerow"
    }
  }

  render() {
    return (
      <div className="navbar-fixed">
        <nav className="white">
          <div className="container nav-wrapper white">
            <Link className='brand-logo left blue-grey-text text-darken-1' to="/">
              {this.state.title}
            </Link>
            <ul className="right hide-on-med-and-down">
              <li>
                <Link className="nav-link blue-grey-text text-darken-1" to="/portfolio">Portfolio</Link>
              </li>
              <li>
                <Link className="nav-link blue-grey-text text-darken-1" to="/about">About</Link>
              </li>
              <li>
                <Link className="nav-link blue-grey-text text-darken-1" to="/resume">Resume</Link>
              </li>
              <li>
                <Link className="nav-link blue-grey-text text-darken-1" to="/contact">Contact</Link>
              </li>
            </ul>
          </div>

        </nav>
      </div>
    );
  }
}

export default Navbar;