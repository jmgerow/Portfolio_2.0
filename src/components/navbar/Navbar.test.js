import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar'

describe('Navbar', () => {
    const navbar = shallow(<Navbar />)
  
    it('renders without crashing', () => {
      expect(navbar).toMatchSnapshot();
    });
 
    it('displays name', () => {
      expect(navbar.state().title).toEqual("Justin Gerow")
    });
  });

