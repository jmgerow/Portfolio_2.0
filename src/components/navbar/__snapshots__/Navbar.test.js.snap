// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="white"
      >
        <div
          className="container nav-wrapper white"
        >
          <Link
            className="brand-logo left blue-grey-text text-darken-1"
            to="/"
          >
            Justin Gerow
          </Link>
          <ul
            className="right hide-on-med-and-down"
          >
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/portfolio"
              >
                Portfolio
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/about"
              >
                About
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/resume"
              >
                Resume
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/contact"
              >
                Contact
              </Link>
            </li>
          </ul>
        </div>
      </nav>,
      "className": "navbar-fixed",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container nav-wrapper white"
        >
          <Link
            className="brand-logo left blue-grey-text text-darken-1"
            to="/"
          >
            Justin Gerow
          </Link>
          <ul
            className="right hide-on-med-and-down"
          >
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/portfolio"
              >
                Portfolio
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/about"
              >
                About
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/resume"
              >
                Resume
              </Link>
            </li>
            <li>
              <Link
                className="nav-link blue-grey-text text-darken-1"
                to="/contact"
              >
                Contact
              </Link>
            </li>
          </ul>
        </div>,
        "className": "white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="brand-logo left blue-grey-text text-darken-1"
              to="/"
            >
              Justin Gerow
            </Link>,
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/portfolio"
                >
                  Portfolio
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/about"
                >
                  About
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/resume"
                >
                  Resume
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/contact"
                >
                  Contact
                </Link>
              </li>
            </ul>,
          ],
          "className": "container nav-wrapper white",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Justin Gerow",
              "className": "brand-logo left blue-grey-text text-darken-1",
              "to": "/",
            },
            "ref": null,
            "rendered": "Justin Gerow",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/portfolio"
                  >
                    Portfolio
                  </Link>
                </li>,
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/about"
                  >
                    About
                  </Link>
                </li>,
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/resume"
                  >
                    Resume
                  </Link>
                </li>,
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/contact"
                  >
                    Contact
                  </Link>
                </li>,
              ],
              "className": "right hide-on-med-and-down",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/portfolio"
                  >
                    Portfolio
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Portfolio",
                    "className": "nav-link blue-grey-text text-darken-1",
                    "to": "/portfolio",
                  },
                  "ref": null,
                  "rendered": "Portfolio",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/about"
                  >
                    About
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "About",
                    "className": "nav-link blue-grey-text text-darken-1",
                    "to": "/about",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/resume"
                  >
                    Resume
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Resume",
                    "className": "nav-link blue-grey-text text-darken-1",
                    "to": "/resume",
                  },
                  "ref": null,
                  "rendered": "Resume",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/contact"
                  >
                    Contact
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Contact",
                    "className": "nav-link blue-grey-text text-darken-1",
                    "to": "/contact",
                  },
                  "ref": null,
                  "rendered": "Contact",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="white"
        >
          <div
            className="container nav-wrapper white"
          >
            <Link
              className="brand-logo left blue-grey-text text-darken-1"
              to="/"
            >
              Justin Gerow
            </Link>
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/portfolio"
                >
                  Portfolio
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/about"
                >
                  About
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/resume"
                >
                  Resume
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/contact"
                >
                  Contact
                </Link>
              </li>
            </ul>
          </div>
        </nav>,
        "className": "navbar-fixed",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container nav-wrapper white"
          >
            <Link
              className="brand-logo left blue-grey-text text-darken-1"
              to="/"
            >
              Justin Gerow
            </Link>
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/portfolio"
                >
                  Portfolio
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/about"
                >
                  About
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/resume"
                >
                  Resume
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link blue-grey-text text-darken-1"
                  to="/contact"
                >
                  Contact
                </Link>
              </li>
            </ul>
          </div>,
          "className": "white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="brand-logo left blue-grey-text text-darken-1"
                to="/"
              >
                Justin Gerow
              </Link>,
              <ul
                className="right hide-on-med-and-down"
              >
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/portfolio"
                  >
                    Portfolio
                  </Link>
                </li>
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/about"
                  >
                    About
                  </Link>
                </li>
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/resume"
                  >
                    Resume
                  </Link>
                </li>
                <li>
                  <Link
                    className="nav-link blue-grey-text text-darken-1"
                    to="/contact"
                  >
                    Contact
                  </Link>
                </li>
              </ul>,
            ],
            "className": "container nav-wrapper white",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Justin Gerow",
                "className": "brand-logo left blue-grey-text text-darken-1",
                "to": "/",
              },
              "ref": null,
              "rendered": "Justin Gerow",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/portfolio"
                    >
                      Portfolio
                    </Link>
                  </li>,
                  <li>
                    <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/about"
                    >
                      About
                    </Link>
                  </li>,
                  <li>
                    <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/resume"
                    >
                      Resume
                    </Link>
                  </li>,
                  <li>
                    <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/contact"
                    >
                      Contact
                    </Link>
                  </li>,
                ],
                "className": "right hide-on-med-and-down",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/portfolio"
                    >
                      Portfolio
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Portfolio",
                      "className": "nav-link blue-grey-text text-darken-1",
                      "to": "/portfolio",
                    },
                    "ref": null,
                    "rendered": "Portfolio",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/about"
                    >
                      About
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "About",
                      "className": "nav-link blue-grey-text text-darken-1",
                      "to": "/about",
                    },
                    "ref": null,
                    "rendered": "About",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/resume"
                    >
                      Resume
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Resume",
                      "className": "nav-link blue-grey-text text-darken-1",
                      "to": "/resume",
                    },
                    "ref": null,
                    "rendered": "Resume",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link blue-grey-text text-darken-1"
                      to="/contact"
                    >
                      Contact
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Contact",
                      "className": "nav-link blue-grey-text text-darken-1",
                      "to": "/contact",
                    },
                    "ref": null,
                    "rendered": "Contact",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
